### PR TITLE
trt-1538: Wait for monitor resources cleanup

### PR DESCRIPTION
**Pull request by jluhrsen**

### CODE REVIEW

File: pkg/cmd/openshift-tests/monitor/run/run_monitor_command.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func (o *RunMonitorOptions) Run() error {
 
 	fmt.Fprintf(o.Out, "Monitor shutting down, this may take up to five minutes...\n")
 
-	cleanupContext, cleanupCancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	cleanupContext, cleanupCancel := context.WithTimeout(context.Background(), 15*time.Minute)
```
Comment (neisw) [inline pkg/cmd/openshift-tests/monitor/run/run_monitor_command.go:189]: I wound up bumping this to 20 minutes [master run_monitor_command](https://github.com/openshift/origin/blob/master/pkg/cmd/openshift-tests/monitor/run/run_monitor_command.go#L189)

File: pkg/monitortests/imageregistry/disruptionimageregistry/monitortest.go [hunk 2]
```diff
@@ -149,12 +152,35 @@ func (w *availability) WriteContentToStorage(ctx context.Context, storageDir, ti
...
+
 func (w *availability) Cleanup(ctx context.Context) error {
 	if w.imageRegistryRoute != nil {
 		err := w.routeClient.RouteV1().Routes("openshift-image-registry").Delete(ctx, w.imageRegistryRoute.Name, metav1.DeleteOptions{})
 		if err != nil {
 			return fmt.Errorf("failed to delete route: %w", err)
 		}
+
+		startTime := time.Now()
+		err = wait.PollUntilContextTimeout(ctx, 15*time.Second, 15*time.Minute, true, w.routeDeleted)
```
Comment (neisw) [inline pkg/monitortests/imageregistry/disruptionimageregistry/monitortest.go:177]: 20

File: pkg/monitortests/network/disruptionpodnetwork/monitortest.go [hunk 3]
```diff
@@ -356,11 +357,34 @@ func (pna *podNetworkAvalibility) WriteContentToStorage(ctx context.Context, sto
...
+}
+
 func (pna *podNetworkAvalibility) Cleanup(ctx context.Context) error {
 	if len(pna.namespaceName) > 0 && pna.kubeClient != nil {
 		if err := pna.kubeClient.CoreV1().Namespaces().Delete(ctx, pna.namespaceName, metav1.DeleteOptions{}); err != nil {
 			return err
 		}
+
+		startTime := time.Now()
+		err := wait.PollUntilContextTimeout(ctx, 15*time.Second, 15*time.Minute, true, pna.namespaceDeleted)
```
Comment (neisw) [inline pkg/monitortests/network/disruptionpodnetwork/monitortest.go:381]: 20

File: pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go [hunk 4]
```diff
@@ -272,12 +275,35 @@ func (w *availability) WriteContentToStorage(ctx context.Context, storageDir, ti
...
+
 func (w *availability) Cleanup(ctx context.Context) error {
 	if len(w.namespaceName) > 0 && w.kubeClient != nil {
 		if err := w.kubeClient.CoreV1().Namespaces().Delete(ctx, w.namespaceName, metav1.DeleteOptions{}); err != nil {
 			return err
 		}
 	}
+
+	startTime := time.Now()
+	err := wait.PollUntilContextTimeout(ctx, 15*time.Second, 15*time.Minute, true, w.namespaceDeleted)
```
Comment (neisw) [inline pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go:300]: and 20 minutes here as well

Comment (jluhrsen) [reply pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go]: fixed

File: pkg/monitortests/imageregistry/disruptionimageregistry/monitortest.go (hunk 2 above)
Comment (jluhrsen) [reply pkg/monitortests/imageregistry/disruptionimageregistry/monitortest.go]: fixed

File: pkg/cmd/openshift-tests/monitor/run/run_monitor_command.go (hunk 1 above)
Comment (jluhrsen) [reply pkg/cmd/openshift-tests/monitor/run/run_monitor_command.go]: fixed

File: pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go [hunk 5]
```diff
@@ -272,12 +275,35 @@ func (w *availability) WriteContentToStorage(ctx context.Context, storageDir, ti
...
+
 func (w *availability) Cleanup(ctx context.Context) error {
 	if len(w.namespaceName) > 0 && w.kubeClient != nil {
 		if err := w.kubeClient.CoreV1().Namespaces().Delete(ctx, w.namespaceName, metav1.DeleteOptions{}); err != nil {
 			return err
 		}
 	}
+
+	startTime := time.Now()
+	err := wait.PollUntilContextTimeout(ctx, 20*time.Second, 15*time.Minute, true, w.namespaceDeleted)
```
Comment (neisw) [inline pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go:300]: I think you meant to bump the timeout (minutes) not the interval

**Comment by jluhrsen** [comment]
@neisw , is there any value in this PR? the purpose of it was to prove that our [OCPBUGS-31868](https://issues.redhat.com//browse/OCPBUGS-31868) is not a regression in 4.16. So assuming it's not a regression, this PR has the chance of making new failures show up in 4.15. Or, we can get this in and backport [the change](https://github.com/openshift/origin/pull/28761) to allow for some errors in the checking of namespaces.

**Comment by neisw** [comment]
@jluhrsen The issue in 4.16 is occurring infrequently I believe.  If you don't have a strong push to get this in and compare the results it seems reasonable to hold off.  We didn't backport it originally even though the root issue that surfaced the need for this was in 4.15.  I don't think it is super risky since we have been running it for a while but if you are ok without it we could close this.

**Comment by jluhrsen** [comment]
totally agree, @neisw. closing this now.

**Pull request closed without merging**